### PR TITLE
info: Show drive/server tolerance

### PR DESCRIPTION
**Pull request by vadmeste**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/admin-info.go [hunk 1]
```diff
@@ -271,8 +305,13 @@ func (u clusterStruct) String() (msg string) {
 	if backendType == madmin.Erasure {
 		msg += "Pools:\n"
 		for pool, summary := range clusterSummary {
-			msg += fmt.Sprintf("   %s, Erasure sets: %d, Drives per erasure set: %d\n",
+			msg += fmt.Sprintf("   %s, Erasure sets: %d, Drives per erasure set: %d",
 				console.Colorize("Info", humanize.Ordinal(pool+1)), summary.setsCount, summary.drivesPerSet)
+			driveTolerance, serverTolerance := poolTolerance(len(summary.endpoints), summary.drivesPerSet, u.Info.StandardParity())
+			msg += fmt.Sprintf(", Drives tolerance: %s, Servers tolerance: %s",
```
Comment (harshavardhana) [inline cmd/admin-info.go:311]:
```suggestion
			msg += fmt.Sprintf(", Drive(s) tolerance: %s, Server(s) tolerance: %s",
```

Comment (harshavardhana) [inline cmd/admin-info.go:311]:
```suggestion
			msg += fmt.Sprintf(", Drive(s) tolerance: %s, Server(s) tolerance: %s",
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-info.go [hunk 2]
```diff
@@ -111,6 +119,32 @@ func endpointToPools(endpoint string, c clusterInfo) (pools []int) {
...
+	if driveFailureTolerance == stripeSize/2 {
+		driveFailureTolerance--
+	}
+
+	serverFailureTolerance := math.Floor((float64(driveFailureTolerance) * float64(numServersPerShard)) / float64(stripeSize))
+
+	return driveFailureTolerance, int(serverFailureTolerance)
+}
+
+func toleranceInfoText(remainingItems int) string {
```
Comment (harshavardhana) [inline cmd/admin-info.go:141]:
```suggestion
func toleranceInfoText(tolerance int) string {
```

File: cmd/admin-info.go [hunk 3]
```diff
@@ -111,6 +119,32 @@ func endpointToPools(endpoint string, c clusterInfo) (pools []int) {
...
+	if remainingItems < 0 {
```
Comment (harshavardhana) [inline cmd/admin-info.go:142]:
```suggestion
	if tolerance < 0 {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by vadmeste** [comment]
I need to review this. This PR doesn't work well when there is more than 16 nodes

**Pull request closed without merging**